modules/broker-core/src/main/java/org/wso2/broker/core/store/dao/MessageDao.java

```diff
@@ -1,38 +1,46 @@
 package org.wso2.broker.core.store.dao;
 
+import org.wso2.broker.core.BrokerException;
 import org.wso2.broker.core.Message;
+import org.wso2.broker.core.store.DbOperation;
+
+import java.util.Collection;
+import javax.sql.DataSource;
 
 /**
  * Defines a functionality required for manipulating messages in persistent storage.
  */
-public interface MessageDao {
+public abstract class MessageDao extends BaseDao {
+
+    public MessageDao(DataSource dataSource) {
+        super(dataSource);
+    }
 
     /**
-     * Storage a message in the persistant storage.
+     * Storage a message in the persistent storage.
      * 
-     * @param message the message.
+     * @param messageList the messages to persist.
      */
-    void persist(Message message);
+    public abstract void persist(Collection<Message> messageList) throws BrokerException;
 
     /**
      * Removes the linkage a messages has with given queue. after the removal if there are links to any other queues
```

Comment:
shall we change as  "Removes a linkage  message"?

Reply:
Shouldn't it be *the* linkage a message has with a given queue.

Reply:
Then we can keep as it is.